projects/pastanaga-angular/src/lib/poc-input/poc-input.component.ts

```diff
@@ -0,0 +1,505 @@
+
+    @Input() set disabled(value: boolean) {
+        this._disabled = coerceBooleanProperty(value);
+    }
+
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    /**** from BaseTextField ****/
+    @Input() set value(value: string | number) {
+        if (!this.isStandalone) {
+            console.warn('Do not use value property and ngModel or form control at the same time');
+        }
+        this.internalPatchValue(value);
+    }
+
+    @Input() set debounceDuration(value: number | undefined) {
+        this._debouncedTime = value || 500;
+        this.debounceTimeChanged.next();
+        this.initDebounce();
+    }
+
+    @Input() placeholder: string = '';
+
+    @Input() set readonly(value: boolean) {
+        this._readonly = coerceBooleanProperty(value);
+    }
+
+    get readonly(): boolean {
+        return this._readonly;
+    }
+
+    @Input() errorMessages?: ErrorMessages;
+    /**
+     * Manual error messaging: a manual 'always false' validator
+     * is added to internal form control
+     */
+    @Input() errorMessage?: string;
+    @Input() required?: boolean;
+    @Input() pattern?: RegExp | string;
+
+    /**** from InputComponent ****/
+    @Input() type: 'text' | 'number' | 'password' | 'email' = 'text';
+
+    @Input() set noAutoComplete(value: boolean) {
+        this._noAutoComplete = coerceBooleanProperty(value);
+    }
+
+    get noAutoComplete(): boolean {
+        return this._noAutoComplete;
+    }
+
+    @Input() set hasFocus(value: boolean) {
+        this._hasFocus = coerceBooleanProperty(value);
+        if (this._hasFocus && !!this.input && this.isActive()) {
+            this.input.nativeElement.focus();
+        }
+    }
+
+    get hasFocus(): boolean {
+        return this._hasFocus;
+    }
+
+    @Input() min?: number;
+    @Input() max?: number;
+    @Input() maxlength?: number;
+    @Input() updateValidatorEvent?: boolean;
+
+    /*** ***/
+
+    @ViewChild('input') input?: ElementRef;
+    @ViewChild('label') labelElement?: ElementRef;
+
+    @Output() debouncedValueChange: EventEmitter<string | number> = new EventEmitter();
+    @Output() focusing: EventEmitter<FocusEvent> = new EventEmitter();
+    @Output() keyUp: EventEmitter<any> = new EventEmitter();
+    @Output() enter: EventEmitter<{ event: KeyboardEvent; value: any }> = new EventEmitter();
+    // Stand alone only
+    @Output() valueChange: EventEmitter<string | number> = new EventEmitter();
+    @Output() blurring: EventEmitter<string | number> = new EventEmitter();
+
+    /*** localHtmlState ***/
+    _id = '';
+    _name = '';
+    _fieldType = 'input';
+    _helpId = '';
+    _label = '';
+    _hasFocus = false;
+    _readonly = false;
+    _noAutoComplete = false;
+    _autofilled = false;
+    _disabled = false;
+
+    /*** internal state ***/
+    internalModel?: string | number = '';
+    internalControl: AbstractControl = new FormControl();
+    isStandalone = true;
+    isFormControl = false;
```

Comment:
is it possible to have `isStandalone` and `isFormControl` both true or both false? Do you really need to keep both of them? (seeing just their names, it feels like the input can be either one or the other)

Reply:
yes, there is another possibility : it's ngModel... so when it's not standalone nor formControl, it's ngModel

we could definitely define a unique identifier like... "mode" and then test if mode === standAlone, if mode === ngModel, if mode === formControl